### PR TITLE
Add HabitTracker Project

### PR DESCRIPTION
**Pull request by arian1917**
Habit Tracker Arian

### CODE REVIEW

**Review by chrisjamiecarter — CHANGES_REQUESTED** [review]
Hey @arian1917,

Before I review you project, you must ensure you have met all the requirements!

Please review:

- [ ] Your project needs to contain a Read Me file where you'll explain how your app works.

Once you have fulfilled this requirement, please **Commit** and **Push** your changes, and I will continue my review.

Also, one quick thing I noticed:

**Solution Structure**
Re-read the [Code Reviews](https://thecsharpacademy.com/article/52/code-reviews) academy page. Specifically step 3.
This will make it easier for code-reviewers to pull down and review your code in the future.
When submitting your code review pull request, all your solution files should be contained inside a new folder as per below:

```
// Current.
.
└── arian1917/
    └── arian1917/
        ├── [otherfiles]
        └── HabitTracker.sln

// Correct.
.
└── HabitTracker.arian1917/
    ├── HabitTracker/
    └── HabitTracker.sln
```

Thanks
@chrisjamiecarter

**Comment by arian1917** [comment]
Ok,  i think i insert the readme. Sorry for all the inconvenience i'm a total incompetent using git and github

File: arian1917/arian1917/Program.cs [hunk 1]
```diff
@@ -0,0 +1,579 @@
+﻿using System.Data;
+using System.Diagnostics.Metrics;
+using System.Globalization;
+using System.Linq.Expressions;
+using System.Numerics;
+using System.Reflection.Metadata.Ecma335;
+using System.Runtime.CompilerServices;
+using Microsoft.Data.Sqlite; 
+
+
```
Comment (chrisjamiecarter) [inline arian1917/arian1917/Program.cs:10]: Whitespace: Try not to include double line breaks. It unnecessarily inflates the line count of your files.

File: arian1917/arian1917/Program.cs [hunk 2]
```diff
@@ -0,0 +1,579 @@
...
+namespace habitTracker{
```
Comment (chrisjamiecarter) [inline arian1917/arian1917/Program.cs:11]: Use file-scoped namespaces to save an indentation level.
Also, Namespaces should be in [Pascal Case](https://www.theserverside.com/answer/Pascal-case-vs-camel-case-Whats-the-difference).

```
// Block-Scoped Namespace
namespace HabitTracker
{
    // Code here.
}
```
vs
```
// File-Scoped Namespace
namespace HabitTracker;
// Code here.
```

File: arian1917/arian1917/Program.cs [hunk 3]
```diff
@@ -0,0 +1,579 @@
...
+        
+        static void Main(string[] args)
+        {
+            InitializeDataBase(); 
+            HabitSelection(); 
+        }
+
+        private static void InitializeDataBase()
+        {
+            using (var connection = new SqliteConnection(connectionString))
```
Comment (chrisjamiecarter) [inline arian1917/arian1917/Program.cs:24]: Use the simpler "Using Declaration" syntax to save an indentation level against the "Using Statement" syntax:
```
// Using Statement
using (var connection = new SqliteConnection(connectionString))
{
    // Code here.
}
```
vs
```
// Using Declaration
using var connection = new SqliteConnection(connectionString);
// Code here.
```

File: arian1917/arian1917/Program.cs [hunk 4]
```diff
@@ -0,0 +1,579 @@
...
+                        Id INTEGER PRIMARY KEY AUTOINCREMENT,
+                        Name TEXT, 
+                        MeasureUnit TEXT, 
+                        TableName TEXT
+                    )";
+
+                tableCmd.ExecuteNonQuery();
+
+                tableCmd.CommandText = 
+                    @$"CREATE TABLE IF NOT EXISTS {tableName} (
```
Comment (chrisjamiecarter) [inline arian1917/arian1917/Program.cs:143]: Note: This is open for [SQL Injection attacks](https://www.stackhawk.com/blog/net-sql-injection-guide-examples-and-prevention/)

Try to use parameterised queries.

**Review by chrisjamiecarter — APPROVED** [review]
Hey @arian1917 ,

Don't sweat the issues with git. I don't know if you are using the cli or not but here is a handy [cheatsheat](https://education.github.com/git-cheat-sheet-education.pdf)

But congratulations on another application in the bag! 🎉 It’s great to see your progress.
As always I have a completed a fair review, and have a few suggestions to help support your growth as a developer:

**Solution Structure**
As already discussed. Ensure your solution struture is correct before submitting:

```
// Correct.
.
└── [ProjectName].[GitHubUserName]/
    ├── [ProjectsDirectories]/
    └── [ProjectName].sln
```

**Style Guide Compliance**
I have added some comments to align you with [Microsoft's style guide](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions) best practices.
These do not need to be actioned now, but should be kept in mind going forward.

Feel free to reach out if you have any questions or if you'd like to collaborate on any of these points.

I will go ahead and mark as approved, keep up the excellent work on the next projects! 😊

Best regards,
@chrisjamiecarter

**Pull request closed without merging**